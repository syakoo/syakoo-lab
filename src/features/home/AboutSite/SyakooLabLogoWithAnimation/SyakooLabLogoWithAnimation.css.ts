import { keyframes, style, type CSSProperties } from "@vanilla-extract/css";

const scaleUp = keyframes({
  "0%": {
    r: "0",
  } as CSSProperties,
  "100%": {
    r: "140",
  } as CSSProperties,
});
export const scaleUpCircleStyle = style({
  animation: `${scaleUp} 0.2s 0.3s forwards ease-out`,
  r: "0",
} as CSSProperties);

const offsetIn = keyframes({
  "0%": {
    strokeDasharray: "0px 227px",
  },
  "100%": {
    strokeDasharray: "227px 0px",
  },
});
export const bGBorderStyle = style({
  animation: `${offsetIn} 0.2s forwards ease-out`,
});

const fadeInOpacity = keyframes({
  "0%": {
    opacity: 0,
  },
  "100%": {
    opacity: 1,
  },
});
export const literalSStyle = style({
  animation: `${fadeInOpacity} 0.3s 0.75s forwards ease-out`,
  opacity: 0,
});

const fadeInAndBounce = keyframes({
  "0%": {
    opacity: 0,
    transform: "translate(0px, -50%)",
  },
  "25%": {
    opacity: 1,
    transform: "translate(0px, 0px) scale(1, 1)",
  },
  "40%": {
    transformOrigin: "calc(100% - 26px) bottom",
    transform: "rotate(7deg)",
  },
  "65%": {
    transform: "translate(0px, 0px)",
  },
  "80%": {
    transformOrigin: "bottom left",
    transform: "rotate(-2deg)",
  },
  "100%": {
    opacity: 1,
  },
});
export const cupStyle = style({
  animation: `${fadeInAndBounce} 0.5s 0.2s forwards cubic-bezier(1, 0.02, 1, 0.68)`,
  transformOrigin: "center",
  opacity: 0,
});

const goAround = keyframes({
  "0%": {
    transform: "rotate(0deg)",
  },
  "40%": {
    transform: "rotate(0deg)",
  },
  "65%": {
    transform: "rotate(2deg)",
  },
  "75%": {
    transform: "rotate(1deg)",
  },
  "80%": {
    transform: "translate(-2px, 2px) rotate(-1deg)",
  },
  "85%": {
    transform: "translate(-1px, 2px) rotate(-1deg)",
  },
  "100%": {
    transform: "translate(0px, 0px) rotate(0deg)",
  },
});
export const cupSpoonPath = style({
  transformOrigin: "center bottom",
  animation: `${goAround} 0.5s 0.2s forwards ease-out`,
});
