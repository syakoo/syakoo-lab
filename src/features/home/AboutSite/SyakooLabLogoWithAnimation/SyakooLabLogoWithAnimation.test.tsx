import { testStories } from "@/test/testStories";

import * as stories from "./SyakooLabLogoWithAnimation.stories";

describe("SyakooLabLogoWithAnimation", () => {
  testStories(stories);
});
