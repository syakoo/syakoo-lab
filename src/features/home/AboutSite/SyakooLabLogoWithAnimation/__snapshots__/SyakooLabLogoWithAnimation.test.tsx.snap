// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyakooLabLogoWithAnimation スナップショット Sample 1`] = `
<body>
  <div>
    <svg
      fill="none"
      height="300"
      viewBox="0 0 120 120"
      xmlns="http://www.w3.org/2000/svg"
    >
      <clippath
        id="syakoo-lab_bg-rect"
      >
        <circle
          class="SyakooLabLogoWithAnimation_scaleUpCircleStyle__8abbm61"
          cx="60"
          cy="120"
          r="140"
        />
      </clippath>
      <path
        clip-path="url(#syakoo-lab_bg-rect)"
        d="M3.15143 3H116.447V116.536H3.15143V3Z"
        fill="white"
      />
      <path
        class="SyakooLabLogoWithAnimation_bGBorderStyle__8abbm63"
        d="M3.15143 3H116.447V116.536H3.15143V3Z"
        stroke="#294E80"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="6"
      />
      <path
        class="SyakooLabLogoWithAnimation_literalSStyle__8abbm65"
        d="M45.7044 5.13562C37.2804 6.06032 29.0241 6.73346 18.2937 11.1178C7.86797 15.969 3.73244 21.4252 2.49012 26.2963C0.837455 35.2134 3.61032 41.3101 7.13297 46.6535C10.941 51.9904 18.4928 53.7062 31.3294 52.1892C38.1573 51.518 48.9005 48.9601 50.883 55.6714C52.5735 61.7838 48.7441 65.0771 44.0973 67.3678L41.9106 68.3622L37.4008 69.4214L64.008 69.2428C67.908 63.1827 69.4537 56.1238 65.5258 46.7428C61.7012 40.2572 58.8071 33.2379 39.2758 35.7606C30.2289 37.0656 19.7636 40.504 17.0437 32.5463C15.1002 25.3602 23.6021 22.3805 29.8115 19.6892C35.8163 17.7018 43.5988 17.4296 50.0794 18.082L45.7044 5.13562Z"
        fill="#454545"
        stroke="#454545"
        stroke-width="0.999999"
      />
      <g
        class="SyakooLabLogoWithAnimation_cupStyle__8abbm67"
      >
        <path
          class="SyakooLabLogoWithAnimation_cupSpoonPath__8abbm69"
          d="M93.8437 47.7807L83.6847 46.0824C83.4123 46.0369 83.1546 46.2208 83.1091 46.4932L74.2338 99.5865C74.1883 99.8588 74.3722 100.117 74.6446 100.162L84.8036 101.86C85.076 101.906 85.3337 101.722 85.3792 101.45L94.2544 48.3562C94.3 48.0839 94.1161 47.8262 93.8437 47.7807Z"
          fill="#294E80"
        />
        <path
          d="M85.4202 95.6494L90.4511 68.6232C91.123 67.6194 92.0166 66.6347 93.1322 65.669C94.2682 64.6888 95.6521 63.948 97.2841 63.4468C98.9365 62.9309 100.892 62.8618 103.151 63.2394C106.094 63.7312 108.679 64.9556 110.908 66.9124C113.14 68.8517 114.743 71.4869 115.717 74.8179C116.693 78.1315 116.795 82.0999 116.022 86.7233C115.269 91.224 113.907 94.9028 111.936 97.7596C109.984 100.602 107.632 102.612 104.879 103.79C102.146 104.953 99.2474 105.279 96.1826 104.766C94.0111 104.403 92.2235 103.735 90.8199 102.763C89.4339 101.793 88.35 100.685 87.5683 99.4379C86.7895 98.1736 86.2399 96.9386 85.9194 95.7329L85.4202 95.6494ZM87.6337 81.9238C87.2326 84.3231 87.2155 86.4715 87.5824 88.369C87.9493 90.2665 88.6797 91.8198 89.7738 93.0288C90.8707 94.2203 92.3211 94.9668 94.1249 95.2683C95.9463 95.5728 97.5708 95.3313 98.9986 94.5439C100.429 93.7389 101.618 92.4974 102.565 90.8195C103.532 89.1269 104.211 87.1073 104.603 84.7606C104.993 82.4314 105.011 80.3282 104.659 78.4511C104.307 76.574 103.584 75.031 102.49 73.822C101.395 72.613 99.929 71.8548 98.0902 71.5474C96.2689 71.243 94.6473 71.4669 93.2253 72.2194C91.8209 72.9747 90.6381 74.1811 89.677 75.8387C88.7158 77.4962 88.0347 79.5246 87.6337 81.9238Z"
          fill="#294E80"
        />
        <path
          d="M88.6848 66.2239L19.6181 66.2239C18.2558 66.2239 17.1514 67.2984 17.1514 68.6238L17.1514 111.824C17.1514 113.149 18.2558 114.224 19.6181 114.224L88.6848 114.224C90.0471 114.224 91.1514 113.149 91.1514 111.824L91.1514 68.6239C91.1514 67.2984 90.0471 66.2239 88.6848 66.2239Z"
          fill="#294E80"
        />
      </g>
    </svg>
  </div>
</body>
`;
